src/util/simplify_expr.cpp

```diff
@@ -236,6 +236,21 @@ bool simplify_exprt::simplify_typecast(exprt &expr)
     return false;
   }
 
+  // casts from boolean to a signed int and back:
+  // (boolean)(int)boolean -> boolean
+  if(expr_type.id()==ID_c_bool &&
+     expr.op0().id()==ID_typecast &&
+     op_type.id()==ID_signedbv)
```

Comment:
I believe this case should be merged into the next case ("// eliminate casts to _Bool").